fix/9559: Fix Classic Product Template on mobile

Sidsector9 opened the pull request:
<!-- Start by describing the changes made in this Pull Request, and the reason for such changes. -->

<!-- Reference any related issues or PRs here -->

Fixes #9559 

<!-- Don't forget to update the title with something descriptive. -->
<!-- If you can, add the appropriate labels -->

### Screenshots

<!-- If your change has a visual component, add a screenshot here. A "before" screenshot would also be helpful. -->



| Before | After |
| ------ | ----- |
|<img width="375" src="https://user-images.githubusercontent.com/4463174/239902859-80d933a7-301a-4465-9d76-0d8824c001fa.png"> |<img width="375" src="https://github.com/woocommerce/woocommerce-blocks/assets/17757960/b08e00d4-f8c5-4276-b55a-f39c936a85c7">|

### Testing

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

<!-- Write these steps from the perspective of a "user" (merchant) familiar with WooCommerce. No need to spell out the steps for common setup scenarios (eg. "Create a product"), but do be specific about the thing being tested. Include screenshots demonstrating expectations where that will be helpful. -->

1. Go to the Site Editor
2. Search for the Product Catalog template
3. Add the Classic product template block
4. Observe the difference between `trunk` and the fix branch

* [ ] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested by users (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

### WooCommerce Visibility

<!-- Check this [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WC core or not (part of the feature plugin or experimental)-->

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Fix: class product template overflow issue on mobile

Sidsector9 committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce-blocks into trunk

Sidsector9 committed:
fix css overflow

Sidsector9 committed:
Merge branch 'trunk' into fix/9559

Sidsector9 committed:
Merge branch 'trunk' into fix/9559

gigitux commented on assets/js/blocks/classic-template/editor.scss:
```diff
@@ -7,8 +7,11 @@
 	display: flex;
 	flex-direction: column;
 	max-width: 900px;
-	width: 400px;
```
Instead of using a prefixed width, we could use `max-width:400`. In this way, the media query isn't necessary.

gigitux reviewed (changes requested):
Thanks for your contribution. I think that we could avoid to use the media query with the approach that I suggested. What do you think? 
Furthermore, I think that we should add some padding to the text 
![image](https://github.com/woocommerce/woocommerce-blocks/assets/4463174/cc7b99cf-f0c1-43c0-a852-5560dbabf7bc)

It would be great if we achieve it without media query.

Sidsector9 committed:
review changes

Sidsector9 committed:
Merge branch 'fix/9559' of github.com:Sidsector9/woocommerce-blocks into fix/9559

Sidsector9 committed:
Merge branch 'trunk' into fix/9559

Sidsector9 committed:
Merge branch 'fix/9559' of github.com:Sidsector9/woocommerce-blocks into fix/9559

Sidsector9 commented:
@gigitux I've addressed the changes and added some more related to tablet view.

### Mobile 

<img width="386" alt="Screenshot 2023-06-16 at 7 32 59 PM" src="https://github.com/woocommerce/woocommerce-blocks/assets/17757960/980a9c45-cbe5-4507-b2be-80d38e63240d">

- Added padding
- Reduced the size of the title on mobile
- Reduced the gap between the title text and icon in mobile


### Tablet (Before)

<img width="781" alt="Screenshot 2023-06-16 at 7 36 03 PM" src="https://github.com/woocommerce/woocommerce-blocks/assets/17757960/415b7f10-fee0-426c-bdc2-ee8a0a3aae6e">

### Tablet (After)

<img width="778" alt="Screenshot 2023-06-16 at 7 35 30 PM" src="https://github.com/woocommerce/woocommerce-blocks/assets/17757960/cc4486bb-e657-4865-8a8d-4f75304b43e7">

gigitux commented on assets/js/blocks/classic-template/editor.scss:
```diff
@@ -51,7 +52,15 @@
 	span {
 		@include font-size(larger);
+
+		@media only screen and (max-width: 480px) {
+			@include font-size(regular, 1.25rem);
+		}
```
Are you sure that it is necessary? I think that we can remove it.

Sidsector9 replied:
On mobile screens, the font size is huge. This media query makes it appropriate for mobile screen sizes.

Sidsector9 replied:
Yes I am sure. On mobile screens, the font size is huge. It wraps it to the next line and looks odd.
This media query makes it appropriate for mobile screen sizes, especially for the ones with lower resolutions.

gigitux commented on assets/js/blocks/classic-template/editor.scss:
```diff
@@ -51,7 +52,15 @@
 
+	@media only screen and (max-width: 480px) {
+		gap: $gap-smaller;
+	}
```
Are you sure that it is necessary? I think that we can remove it.

Sidsector9 replied:
@gigitux same as above, the amount of gap was a lot between the icon and the title (after reducing font size) for mobile. Do you think it is unnecessary?

gigitux reviewed (changes requested):
Thanks for working on this PR. I don't see any benefit to using a media query for the font and the gap. The UI looks okay without them. What do you think?

Sidsector9 committed:
Merge branch 'trunk' into fix/9559

gigitux reviewed (approved):
Thanks for working on this! I tried different approaches without success, so I think the media query is the way to go. LGTM!

gigitux commented:
Thank you so much for your contribution! However, due to changes in the design of this block (#9602), I'm unable to merge your PR.

I understand that this may be disappointing, especially considering the effort you put into it. Your work is valued, and I encourage you to continue contributing to the project. I would be more than happy to help you find another task.
Again, thank you for your understanding!
